keel-retrofit/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/AuthorizationHeadersInterceptor.kt

```diff
@@ -18,13 +18,16 @@ class AuthorizationHeadersInterceptor(private val fiatPermissionEvaluator: FiatP
   override fun intercept(chain: Chain): Response {
     var request = chain.request()
     request.header(Header.USER.header)?.also { user ->
-      AuthenticatedRequest.allowAnonymous {
-        val accounts = fiatPermissionEvaluator.getPermission(user).accounts.joinToString(",") { it.name }
-        log.trace("Adding X-SPINNAKER-ACCOUNTS: $accounts to ${request.method} ${request.url}")
-        request = request
-          .newBuilder()
-          .addHeader(Header.ACCOUNTS.header, accounts)
-          .build()
+      AuthenticatedRequest.setUserOrigin("keel")
+      AuthenticatedRequest.propagate {
```

Comment:
Glad you were able to hook this into this interceptor. It didn't exist until last week. 😄